File: app/src/main/java/com/perflyst/twire/tasks/GetStreamsCountTask.java
package com.perflyst.twire.tasks;

import android.content.Context;

import com.google.common.collect.Lists;
import com.perflyst.twire.TwireApplication;
import com.perflyst.twire.model.UserInfo;

import java.lang.ref.WeakReference;
import java.util.concurrent.Callable;

/**
 * Created by Sebastian Rask on 26-06-2016.
 */
public class GetStreamsCountTask implements Callable<Integer> {
    private final WeakReference<Context> context;

    public GetStreamsCountTask(Context context) {
        this.context = new WeakReference<>(context);
    }

    public Integer call() {
        try {
            GetFollowsFromDB subscriptionsTask = new GetFollowsFromDB(context.get());
            var follows = subscriptionsTask.call().values().stream().map(UserInfo::getUserId).toList();

            int total = 0;
            for (var chunk : Lists.partition(follows, 100)) {
                var streams = TwireApplication.helix.getStreams(null, null, null, 100, null, null, chunk, null).execute().getStreams();
                total += streams.size();
            }

            return total;
        } catch (Exception e) {
            e.printStackTrace();
        }
        return -1;
    }
}

File: app/src/main/java/com/perflyst/twire/tasks/GetStreamsCountTask.kt
package com.perflyst.twire.tasks

import android.content.Context
import com.perflyst.twire.TwireApplication
import timber.log.Timber
import java.lang.ref.WeakReference
import java.util.concurrent.Callable

/**
 * Created by Sebastian Rask on 26-06-2016.
 */
class GetStreamsCountTask(context: Context?) : Callable<Int> {
    private val context: WeakReference<Context?> = WeakReference<Context?>(context)

    override fun call(): Int {
        try {
            val subscriptionsTask = GetFollowsFromDB(context.get())
            val follows = subscriptionsTask.call().values.map { it.userId }

            return follows.chunked(100).flatMap {
                TwireApplication.helix.getStreams(
                    null,
                    null,
                    null,
                    100,
                    null,
                    null,
                    it,
                    null
                ).execute().streams
            }.size
        } catch (e: Exception) {
            Timber.e(e)
        }
        return -1
    }
}
